hack/update-codegen.sh

```diff
@@ -657,6 +638,10 @@ function codegen::clients() {
         done
     fi
 
+    git_find -z 'staging/src/k8s.io/client-go/**/*.go' \
```

Comment:
That would be cleaner if we could follow that pattern but 2 odd cases:

1) applyconfig has a manual top-level doc.go - maybe they all should (applyconfig, listers, informers, clients) instead of generating it?  Then it's not a one-off exclusion any more.

Some of the subprojects have "deprecated" trees that are not regenerated - I can't tell if those should be removed or un-tagged as "generated by"?  E.g. staging/src/k8s.io/apiextensions-apiserver/pkg/client/clientset/deprecated/...

Reply:
> applyconfig has a manual top-level doc.go - maybe they all should (applyconfig, listers, informers, clients) instead of generating it? Then it's not a one-off exclusion any more.

+1 to consistent doc.go file to make the package resolvable to go tools even when all the generated files are removed

> Some of the subprojects have "deprecated" trees that are not regenerated - I can't tell if those should be removed or un-tagged as "generated by"? E.g. staging/src/k8s.io/apiextensions-apiserver/pkg/client/clientset/deprecated/...

probably remove them... I think they got missed in #95923

Reply:
Will remove deprecated.